### PR TITLE
Print inlined functions on Windows

### PR DESCRIPTION
**Pull request by Zoxc**
Split from https://github.com/rust-lang/rust/pull/45637

r? @alexcrichton

### CODE REVIEW

File: src/libstd/sys_common/backtrace.rs [hunk 1]
```diff
@@ -100,16 +103,25 @@ fn filter_frames(frames: &[Frame],
         return (0, 0);
     }
 
-    let skipped_before = 0;
+    // Look for the first occurence of a panic entry point
+    // Skip all frames before that
+    let skipped_before = frames.iter().position(|frame| {
```
Comment (alexcrichton) [inline src/libstd/sys_common/backtrace.rs]: Did these leak in by accident? I think tidy failed to compile due to these maybe?

**Comment by alexcrichton** [comment]
Can [this test](https://github.com/rust-lang/rust/blob/1b193de98af5b8c6e2b50908ea64323b9095a1a4/src/test/run-pass/backtrace-debuginfo.rs#L87-L114) also be updated?

File: src/libstd/sys_common/backtrace.rs (hunk 1 above)
Comment (Zoxc) [reply src/libstd/sys_common/backtrace.rs]: I forgot to commit my code...

**Comment by alexcrichton** [comment]
Was it possible to update the relevant test to avoid the special-casing of msvc?

**Comment by Zoxc** [comment]
It looks like location information disappeared from x86-32. I'll have to look into that.

**Comment by kennytm** [comment]
Hi @Zoxc! Have you checked the issue of 32-bit x86?

**Comment by Zoxc** [comment]
I checked that similar code does work for C++. There might be some issue with either the FFI or debug information Rust generates. I'll try to use the C++ code to generate a stack trace for Rust code.

**Comment by Zoxc** [comment]
I used a pointer instead of a `u64` in `SymFromInlineContext`, which messed up the ABI on 32-bit platforms.

**Comment by Zoxc** [comment]
This is now ready to be merged. The backtrace test still doesn't pass because the debug information we emit is not entirely correct.

**Comment by alexcrichton** [comment]
@bors: r+

Ok, thanks for investigating!

**Comment by kennytm** [comment]
`dist-various-2` failed, cannot build on CloudABI, legit.

```
[00:47:16] warning: dropping unsupported crate type `dylib` for target `x86_64-unknown-cloudabi`
[00:47:16] 
[00:47:19] error[E0063]: missing field `inline_context` in initializer of `sys_common::backtrace::Frame`
[00:47:19]   --> libstd/sys/cloudabi/backtrace.rs:77:29
[00:47:19]    |
[00:47:19] 77 |         cx.frames[cx.idx] = Frame {
[00:47:19]    |                             ^^^^^ missing `inline_context`
[00:47:19] 
[00:47:21] error: aborting due to previous error
[00:47:21] 
[00:47:21] error: Could not compile `std`.
```

**Comment by alexcrichton** [comment]
Gonna merge anyway since all tests passed and it's just the appveyor timeout.

**Pull request merged**